### PR TITLE
Add NumPy + Client API example

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
### Description

Adds different NumPy based mock training example to showcase different configurations of Client API.
Also showcase how to launch_once and use a training loop in custom/user training script.

- Add README of different configurations options of NVFlare Client API
- Add "sag_np" job template 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/hello-world/ml-to-fl/np/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+In that case, we could use the "launch_once" option of "LauncherExecutor" and wraps our training script into a loop.
+
+We wrap the [./code/train_full.py](./code/train_full.py) into a loop: [./code/train_loop.py](./code/train_loop.py)
+
+Then we can create the job:
+
+```bash
+nvflare job create -force -j ./jobs/np_loop -w sag_np -sd ./code/ \
+-f config_fed_client.conf app_script=train_loop.py params_transfer_type=DIFF launch_once=true \
```
Comment (SYangster) [inline examples/hello-world/ml-to-fl/np/README.md:102]: `params_transfer_type` and `expected_data_kind` should be `FULL` and `WEIGHTS` for train_loop.py

Comment (YuanTingHsieh) [reply examples/hello-world/ml-to-fl/np/README.md]: @SYangster good catch! fixed.

**Review by SYangster — APPROVED** [review]

**Pull request merged**